package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua

```diff
@@ -61,42 +61,31 @@ if wireless.device_supports_mfp(uci) then
 	mfp:value("1", translate("Optional"))
 	mfp:value("2", translate("Required"))
 end
-mfp.default = uci:get('wireless', primary_iface, 'ieee80211w') or "0"
-
+mfp.default = uci:get('gluon', 'wireless', 'private_mfp') or "0"
 
 function f:write()
-	wireless.foreach_radio(uci, function(radio, index)
-		local radio_name = radio['.name']
-		local suffix = radio_name:match('^radio(%d+)$')
-		local name   = "wan_" .. radio_name
-
-		if enabled.data then
-			local macaddr = wireless.get_wlan_mac('wan_radio', index, radio)
-
-			uci:section('wireless', 'wifi-iface', name, {
-				device     = radio_name,
-				network    = 'wan',
-				mode       = 'ap',
-				encryption = encryption.data,
-				ssid       = ssid.data,
-				key        = key.data,
-				macaddr    = macaddr,
-				ifname     = suffix and 'wl-wan' .. suffix,
-				disabled   = false,
-			})
+	if enabled.data then
+		-- set new private wifi configuration
+		uci:set('gluon', 'wireless', 'private_ssid', ssid.data)
+		uci:set('gluon', 'wireless', 'private_key', key.data)
+		uci:set('gluon', 'wireless', 'private_encryption', encryption.data)
+		uci:set('gluon', 'wireless', 'private_mfp', mfp.data)
+	end
 
-			-- hostapd-mini won't start in case 802.11w is configured
-			if wireless.device_supports_mfp(uci) then
-				uci:set('wireless', name, 'ieee80211w', mfp.data)
+	for _, band in ipairs({'band_2g', 'band_5g'}) do
```

Comment:
This must iterate over the existing bands in the wireless config, then you can drop the `device_supports_band` here.